R/reporter.R

```diff
@@ -18,6 +18,24 @@ Reporter <- R6::R6Class("Reporter",
     end_context =    function(context) {},
     end_reporter =   function() {},
 
+    out = NULL,
+
+    initialize = function() {
+      self$out <- stdout()
+    },
+
+    cat = function(..., file = NULL, sep = " ", fill = FALSE, labels = NULL,
+                   append = NULL) {
+      if (!is.null(file)) {
+        warning("file ignored", call. = FALSE)
+      }
+      if (!is.null(append)) {
+        warning("append ignored", call. = FALSE)
+      }
+
+      cat(..., file = self$out, sep = sep, fill = fill, labels = labels)
```

Comment:
I'd like to tackle cat() refactoring in a separate PR, this is already big enough.

Caveat: Not all cat() calls have sep = "" in place, but the integration tests should catch this.